### PR TITLE
Module 11

### PR DESCRIPTION
**Pull request by jaybee1971**
Had to do a lot of refactoring (as advised in exercise 10) to get this ready for Azure, but all seems to be working now.  There is one slight issue with the OAuth callback not going back to the root index and it looking like you don't have writer permissions.  If you manually change the URL or relaunch the app it allows full access.  I couldn't work out how to fix this, hopefully quite simple.

### CODE REVIEW

File: todo_app/app.py [hunk 1]
```diff
@@ -24,47 +24,50 @@ def create_app():
...
     @login_required
     def new_todo():
-        if current_user.role == "writer":
+        if app.config["LOGIN_DISABLED"] == "True" or current_user.role == "writer":
             last_update = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%fZ')
-            mongo_post(request.form['add_todo'], request.form['add_desc'], request.form['due_date'], last_update)
+            database_post(request.form['add_todo'], request.form['add_desc'], request.form['due_date'], last_update)
             app.logger.info('Processing create new card request')
-            return redirect('/')
+            return index()
```
Comment (hugh-emerson) [inline todo_app/app.py:47]: Why did you change this?
`return redirect('/')` is better than `return index()`

It will update the browser's URL correctly, and the user can hit refresh without resubmitting the form.

And by getting the browser to make a fresh request, you will avoid potential problems if the index function tries to look at the current request.

**Review by hugh-emerson — APPROVED** [review]
Did you create the app with your own Azure subscription?

We expected you to use the `AmericanExpress1_JasonBuckenham_ProjectExercise` resource group in the Softwire Academy directory. Means we can see them and also we want to make sure you're not paying for anything!

You can switch over in the module 12 exercise when you create the resources via Terraform.

Otherwise, looks good. Just one minor comment about the redirects. If you're struggling to get the behaviour you want, then please send us a message

Merge when you're ready 👍

File: todo_app/app.py (hunk 1 above)
Comment (jaybee1971) [reply todo_app/app.py:47]: I'll change it back, I was trying to see if it helped with the redirect issue I was having during the exercise.  It didn't but everything seemed to work the same so I left the update in.
